int8 mode
string[] color
---
bool ack